_posts/2020-03-06-introducing-mirrormaker2.md

```diff
@@ -0,0 +1,285 @@
+Essentially, you are limited to _active/passive_ synchronization.
+MirrorMaker 2.0 solves this by introducing the concept of _remote_ topics.
+
+_Remote_ topics are created from the originating cluster by the `MirrorSourceConnector`.
+They are distinguished by automatic renaming that appends the name of cluster to the name of the topic.
+Our _Topic-1.Partition-1_ from the source cluster becomes the never-to-be-confused _Cluster-1-Topic-1.Partition-1_ in the target cluster.
+
+As you can see here, _remote_ topics can be easily identified so that they are not returned to the source cluster.
+
+![Topic renaming with MirrorMaker 2.0](/assets/2020-03-12-mirrormaker-renaming.png)
+
+Why _remote_ topic and not just _source_ topic?
+Because each source cluster can also be a target cluster in a bidirectional configuration.
+This feature also means that it is no longer necessary to aggregate data in a separate cluster.
+
+A rose might be a rose by any other name, but when it comes to replicating topics in MirrorMaker 2.0 we can't afford such ambiguity.
+Particularly if we want to create a bidirectional configuration.
+
+The approach to topic renaming opens up a world of bidirectional opportunities.
+You can now have an _active/active_ cluster configuration that feeds data to and from each cluster.
+For this, you need a MirrorMaker 2.0 cluster at each destination.
+
+### Self-regulating topic replication
+
+Topic replication is defined using regular expression patterns to _whitelist_ or _blacklist_ topics:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker2
+spec:
+  # ...
+  mirrors:
+  - sourceCluster: "my-cluster-source"
+    targetCluster: "my-cluster-target"
+    topicsPattern: ".*"
+    # ...
+```
+
+You use `topicsBlacklistPattern` if you want to use blacklists.
+
+Pretty similar to before, but previously topic _configuration_ was not replicated.
+The target cluster had to be manually maintained to match topic configuration changes in the source cluster.
+
+This time the topic configuration is automatically synchronized between source and target clusters through configuration properties.
+Configuration changes are propagated to remote topics so that new topics and partitions are detected and created.
+The need for rebalancing vanishes.
+
+### Offset tracking and mapping
+
+In the old version of MirrorMaker, the offset of the source topic in the target cluster begins when the replication begins.
+The `__consumer_offsets` topic is not mirrored.
+So offsets of the source topic and its replicated equivalent can have two entirely different positions.
+This was often problematic in a failover situation.
+How to find the offset in the target cluster?
+Strategies such as using timestamps can be adopted, but it adds complexity.
+
+Poof! These issues entirely disappear with MirrorMaker 2.0.
```

Comment:
```suggestion
These issues entirely disappear with MirrorMaker 2.0.
```

Reply:
Not Poof! That was my favourite word in the post. :disappointed:  Removed.